wlauto/workloads/googleplaybooks/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,362 @@
+
+    public static String TAG = "uxperf_googleplaybooks";
+
+    public Bundle parameters;
+    private int viewTimeoutSecs = 10;
+    private long viewTimeout =  TimeUnit.SECONDS.toMillis(viewTimeoutSecs);
+    private LinkedHashMap<String, Timer> timingResults = new LinkedHashMap<String, Timer>();
+
+    public void runUiAutomation() throws Exception {
+        this.timeout = TimeUnit.SECONDS.toMillis(8);
+
+        parameters = getParams();
+
+        String bookTitle = parameters.getString("book_title").replace("_", " ");
+        String searchWord = parameters.getString("search_word");
+        String noteText = "This is a test note";
+
+        setScreenOrientation(ScreenOrientation.NATURAL);
+        dismissSync();
+
+        openMyLibrary();
+        searchForBook(bookTitle);
+
+        selectBook(3);
+        gesturesTest();
+        selectRandomChapter();
+        addNote(noteText);
+        removeNote();
+        searchForWord(searchWord);
+        switchPageStyles();
+        aboutBook();
+
+        pressBack();
+        unsetScreenOrientation();
+
+        writeResultsToFile(timingResults, parameters.getString("output_file"));
+    }
+
+    private void dismissSync() throws Exception {
+        UiObject keepSyncOff =
+            new UiObject(new UiSelector().textContains("Keep sync off")
+                                         .className("android.widget.Button"));
+        if (keepSyncOff.exists()) {
+            keepSyncOff.click();
+        }
+    }
+
+    private void openMyLibrary() throws Exception {
+        Timer result = new Timer();
+        result.start();
+
+        UiObject openDrawer = getUiObjectByDescription("Show navigation drawer",
+                                                       "android.widget.ImageButton");
+        openDrawer.click();
+
+        // To correctly find the UiObject we need to specify the index also here
+        UiObject myLibrary =
+            new UiObject(new UiSelector().className("android.widget.TextView")
+                                         .text("My library").index(3));
+        myLibrary.clickAndWaitForNewWindow(timeout);
+
+        result.end();
+
+        timingResults.put("open_library", result);
+    }
+
+    private void searchForBook(final String text) throws Exception {
+        Timer result = new Timer();
+        result.start();
+        UiObject search =
+            new UiObject(new UiSelector().resourceId("com.google.android.apps.books:id/menu_search"));
+        search.click();
+
+        UiObject searchText =
+            getUiObjectByResourceId("com.google.android.apps.books:id/search_src_text",
+                                    "android.widget.EditText");
+
+        searchText.setText(text);
+        pressEnter();
+
+        UiObject resultList =
+            new UiObject(new UiSelector().resourceId("com.android.vending:id/search_results_list"));
+
+        if (!resultList.waitForExists(viewTimeout * 2)) {
```

Comment:
Could we have a differently named timeout specific to this case rather than *2 for a value used for something else?  What if we decide that a view only needs a timeout of 3 seconds?